python/taichi/aot/utils.py

```diff
@@ -2,11 +2,24 @@
 from taichi.lang.enums import Layout
 from taichi.lang.exception import TaichiCompilationError
 from taichi.lang.matrix import Matrix, MatrixNdarray, MatrixType, VectorNdarray
+from taichi.types import f32, i32
 from taichi.types.annotations import template
 from taichi.types.ndarray_type import NdarrayType
 
 template_types = (NdarrayType, template)
 
+_aliased_type = {i32: int, int: i32, f32: float, float: f32}
+
+
+def check_type_match(lhs, rhs):
```

Comment:
This is not accurate - try searching for `cook_dtype` (sorry naming is not that helpful :P

Reply:
Thanks!